### PR TITLE
Add CodeQL Analysis workflow

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
mirror #1416

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: "Code Scanning - Action"
+
+on:
+  push:
+  schedule:
+    - cron: '0 0 * * 0'
```
Comment (Jamminroot) [inline .github/workflows/codeql.yml:6]: any specific reason to use cron scheduling?

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+jobs:
+  CodeQL-Build:
+
+    strategy:
+      fail-fast: false
```
Comment (Jamminroot) [inline .github/workflows/codeql.yml:12]: I think this task should immediately stop and report fail if something goes wrong, so using `fail-fast: true` might be more appropriate.

File: sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java [hunk 3]
```diff
@@ -319,7 +319,7 @@ protected static void trackPage(String name, Map<String, String> properties) {
      * Additional validation rules apply depending on the configured secret.
      * <p>
      * For App Center, the name cannot be longer than 256 and is truncated otherwise.
-     * For One Collector, the name needs to match the <tt>[a-zA-Z0-9]((\.(?!(\.|$)))|[_a-zA-Z0-9]){3,99}</tt> regular expression.
+     * For One Collector, the name needs to match the <code>[a-zA-Z0-9]((\.(?!(\.|$)))|[_a-zA-Z0-9]){3,99}</code> regular expression.
```
Comment (Jamminroot) [inline sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java:322]: Im not sure - what are those changes?

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (MatkovIvan) [reply .github/workflows/codeql.yml:12]: @Jamminroot we need to get full error list here

File: sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java (hunk 3 above)
Comment (MatkovIvan) [reply sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java:322]: Javadoc errors
`error: tag not supported in the generated HTML version: tt`

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (Jamminroot) [reply .github/workflows/codeql.yml:12]: This one resolved.

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (AnastasiaKubova) [reply .github/workflows/codeql.yml:6]: We have the same implementation in another our PR
https://github.com/microsoft/appcenter-sdk-react-native/pull/826
I think we need to keep uniformity anywhere.

**Pull request merged**